pkg/server/download.go

```diff
@@ -355,3 +355,24 @@ func (s *server) updateDownloadSubscribers(total int64, current int64, name stri
 		s.subscribers.Delete(id)
 	}
 }
+
+// If the source and destination are on different file systems this will cause
+// an invalid cross-link device error using os.Rename. Instead, open the file
+// and then write it out
+func fileSystemSafeRename(tempPath string, targetLocation string) error {
+	byteData, err := os.ReadFile(tempPath)
+	if err != nil {
+		return err
+	}
+	err = os.WriteFile(targetLocation, byteData, 0644)
```

Comment:
why 0644 specifically and not whatever the permissions on the tempfile were? is this what os.Rename does?